A script to auto generate schemas on a side.

zorge69 opened the pull request:
## Pull Request submission

Added a little script to automatically create a toml schema from a csv file,
The script sits on a side, and does not need to be packaged.
It reads a csv in pandas, gets the data types automatically, and then creates a string
with column names (new are the same as old) and data types.

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #


#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)


#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

zorge69 committed:
A script to auto generate schemas on a side.

JenCheshire reviewed (commented):
Looks good @zorge69 - couple of questions:
It's outputting the index in the toml file - is that going to cause any issues?
Debugging/test files inputs are still in the file -  do they need to be taken out?

zorge69 committed:
Schema is written to config/output_schemas

zorge69 committed:
Tidied up debugging and flake8

zorge69 committed:
Removed column index from the schema

zorge69 commented:
> Looks good @zorge69 - couple of questions: It's outputting the index in the toml file - is that going to cause any issues? Debugging/test files inputs are still in the file - do they need to be taken out?

Good points!
1. Dropped column "index" from the schema
2. Removed debugging bits
3. Saving the schema to config/output_schemas

Input file name is still in the script. It's a one-off thing, I don't need it to be parametrised for now.
